generator/lib/src/code_chunks.dart

```diff
@@ -331,24 +335,25 @@ class CodeChunks {
       switch (p.type) {
         case OBXPropertyType.ByteVector:
           if (['Int8List', 'Uint8List'].contains(p.fieldType)) {
-            // No need for the eager reader here. We need to call fromList()
-            // constructor anyway - there's no Int8List.generate() factory.
-            fbReader = 'fb.ListReader<int>(fb.Int8Reader())';
-            if (p.fieldIsNullable) {
-              preLines.add('final $valueVar = ${readFieldOrNull()};');
-              return '$valueVar == null ? null : ${p.fieldType}.fromList($valueVar)';
-            } else {
-              return '${p.fieldType}.fromList(${readFieldNonNull('[]')})';
-            }
+            fbReader = 'fb.${p.fieldType}Reader(lazy: false)';
+            return '${readField()} as ${p.fieldType}${p.fieldIsNullable && p.entity!.nullSafetyEnabled ? "?" : ""}';
           } else {
             fbReader = 'fb.ListReader<int>(fb.Int8Reader(), lazy: false)';
           }
           break;
         case OBXPropertyType.Relation:
           fbReader = 'fb.${_propertyFlatBuffersType[p.type]}Reader()';
           return readFieldNonNull('0');
+        case OBXPropertyType.String:
+          // TODO `asciiOptimization: true` is for keeping the same behavior as the previous FB fork.
+          //      Check if it still makes sense with the latest Dart/Flutter.
```

Comment:
Were these TODOs resolved? Or still needs to be done, e.g. create an issue?

Reply:
Created an issue #360